ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/support/ArticleParser.java

```diff
@@ -0,0 +1,233 @@
+                        jsonObjectHandler(returnedObject);
+                    } else {
+                        // Get inner Article Object
+                        parseArticleObject(returnedObject);
+                    }
+                } catch (Exception e) {
+                    Timber.e(e, "getArticleFromCategory: ");
+                }
+            }
+        }
+    }
+
+    /**
+     * This method parses an Article Json Object into the Database. On completion it will add
+     * articles to <code>articleList</code>
+     *
+     * @param articleObject = article JsonObject
+     * @throws Exception =
+     */
+    private void parseArticleObject(JSONObject articleObject) throws Exception {
+        // Create Article
+        Article retrievedArticle = new Article();
+        // get Inner article Object
+        Iterator<String> keys = articleObject.keys();
+        JSONObject articleTagObject = null;
+        while (keys.hasNext()) {
+            String nextKey = keys.next();
+            if (isArticleOrCategory(nextKey)) {
+                articleTagObject = articleObject.getJSONObject(nextKey);
+            }
+        }
+
+        if (articleTagObject == null) {
+            throw new Exception("Article not configured properly");
+        }
+        retrievedArticle.mDateCreated = getDateLongFromJsonString(articleObject.getString(CREATED_TAG));
+        if (articleObject.has(CONTENT_TAG) && articleObject.getJSONObject(CONTENT_TAG)
+                .has(LAST_MODIFIED_TAG)) {
+            retrievedArticle.mDateUpdated = getDateLongFromJsonString(articleObject
+                    .getJSONObject(CONTENT_TAG).getString(LAST_MODIFIED_TAG));
+        }
+        retrievedArticle.mkey = articleObject.getJSONObject(CONTENT_TAG).getString(UUID_TAG);
+        retrievedArticle.mTitle = articleTagObject.getJSONObject(CONTENT_TAG)
+                .getString(TITLE_TAG);
+
+        JSONObject articleRootObject = articleTagObject.getJSONObject(CONTENT_TAG)
+                .getJSONObject(ROOT_TAG);
+
+        retrievedArticle.mContent = articleRootObject.getJSONObject("text").getString("text");
+
+        // add retrieved Article to List
+        articleList.add(retrievedArticle);
+
+        // get Attachments from Articles
+        getAttachmentsFromRootObject(articleRootObject, retrievedArticle.mkey);
+
+
+    }
+
+    /**
+     *  This method if for extracting Attachments from the root Json Object of the article.  On
+     *  completion it will add Attachment to <code>attachmentList</code>
+     *
+     * @param articleRootObject = the root json Object of Article
+     * @param articleKey = the uuid of the article
+     */
+    private void getAttachmentsFromRootObject(JSONObject articleRootObject, String articleKey) {
+        // Iterate through keys
+        Iterator<String> keys = articleRootObject.keys();
+        while (keys.hasNext()) {
+            String nextKey = keys.next();
+            if (nextKey.contains("image")) {
```

Comment:
you shouldn't be looking for images based on the key name, you should look based on this attribute:
```
"sling:resourceType": "wcm/foundation/components/image"
```